test/puppetlabs/puppetdb/admin_test.clj

```diff
@@ -26,26 +27,84 @@
             [puppetlabs.puppetdb.testutils.services :as svc-utils]
             [puppetlabs.puppetdb.scf.storage :as scf-storage]
             [puppetlabs.puppetdb.testutils.catalog-inputs :refer [sample-input-cmds]]
-            [puppetlabs.puppetdb.time :refer [now days ago] :as time]))
+            [puppetlabs.puppetdb.time :refer [now days ago] :as time])
+  (:import
+    (java.time.format DateTimeFormatter)
+    (java.time ZonedDateTime ZoneId)))
 
 (use-fixtures :each tu/call-with-test-logging-silenced)
 
+(defn update-timestamp [resource resource-list time]
+  (let [current-time  (time/to-date-time time)
+        index (.indexOf resource-list resource)
+        ; We need to preserve the order of the reports in example/reports
+        minus (time/minus current-time (time/parse-period (str (- 10 index) "s")))
+        timestamp (time/unparse (time/formatters :date-time) minus)
+        ]
+    (-> resource
+        (assoc :timestamp timestamp)
+        (update :events (fn [events]
+                                   (mapv #(assoc % :timestamp timestamp)
+                                         events)))
+        )
+    )
+  )
+
+(defn change-report-time
+  ([r time]
+  (change-report-time r time time))
+  ([r time time2]
+  ;; A *very* blunt instrument, only intended to work for now on
+  ;; example/reports.
+  (-> (assoc r
+             :producer_timestamp time2
+             :start_time time
+             :end_time time)
+      (update :logs #(mapv (fn [entry] (assoc entry :time time)) %))
+      (update :resources
+              (fn [resources]
+                (mapv
+                 (fn [res]
+                   (-> res
+                       (update-timestamp resources time)
+                 )) resources))))))
+
+(defn expected-report [current-time plan-report-ts]
+  (let [report-list (filter-reports [:and
+                   [:= :certname example-certname]
+                   [:null? :type false]])
+        timestamp (time/unparse (time/formatters :date-time) current-time)
+        expected-report (mapv (fn [res]
+                                (if (= (get res :type) "plan")
+                                  (change-report-time res timestamp plan-report-ts)
+                                  (change-report-time res timestamp)
+                                  )
+                                )
+                              report-list)]
+    expected-report
+    )
+  )
+
 (deftest test-basic-roundtrip
+  (db/with-test-db
```

Comment:
I'm not sure if we need this `with-test-db` wrapper. Did you notice the tests doing something unexpected without it?